lib/server.js

```diff
@@ -24,13 +24,27 @@ function handleResponse (opts, req, resp, response) {
             response.headers = {};
         }
 
+        var rh = response.headers;
+
         // Set up CORS
-        response.headers['Access-Control-Allow-Origin'] = '*';
-        response.headers['Access-Control-Allow-Methods'] = 'GET';
-        response.headers['Access-Control-Allow-Headers'] = 'accept, content-type';
+        rh['Access-Control-Allow-Origin'] = '*';
+        rh['Access-Control-Allow-Methods'] = 'GET';
+        rh['Access-Control-Allow-Headers'] = 'accept, content-type';
+
+        if (!/^application\/json/.test(rh['content-type'])) {
+            // Set up security headers
+            // https://www.owasp.org/index.php/List_of_useful_HTTP_headers
+            rh['X-XSS-Protection'] = '1; mode=block';
+            rh['X-Content-Type-Options'] = 'nosniff';
+            rh['X-Frame-Options'] = 'SAMEORIGIN';
+            rh['Content-Security-Policy'] =
+                "default-src 'none'; media-src *; img-src *; style-src *";
```

Comment:
I did miss that. So for non json content, is there any case not covered by 'self' and *.wikimedia.org for style/img/media?

For the json case, you should still send nosniff and XFO: DENY. That will help with xssi. There are probably not cases where a browser will ignore nosniff but enfore CSP, but I think the question should be why not include them. If there's a valid reason, then leave it off.